src/states/adult/mod.rs

```diff
@@ -539,69 +462,119 @@ impl Base for Adult {
+                request.into_inner(),
+                msg.src.to_sender_node(sender)?,
                 outbox,
             ),
-            BootstrapRequest(name) => {
-                self.handle_bootstrap_request(p2p_node, name);
-                Ok(Transition::Stay)
-            }
-            ConnectionResponse => {
-                debug!("{} - Received connection response from {}", self, p2p_node);
+            Variant::ParsecResponse(version, response) => self.handle_parsec_response(
+                version,
+                response.into_inner(),
+                *msg.src.as_node()?,
+                outbox,
+            ),
+            Variant::BootstrapRequest(name) => {
+                self.handle_bootstrap_request(msg.src.to_sender_node(sender)?, name);
                 Ok(Transition::Stay)
             }
-            msg @ BootstrapResponse(_) => {
+            Variant::ConnectionResponse => {
                 debug!(
-                    "{} Unhandled direct message from {}, discard: {:?}",
+                    "{} - Received connection response from {}",
                     self,
-                    p2p_node.public_id(),
-                    msg
+                    msg.src.to_sender_node(sender)?
                 );
                 Ok(Transition::Stay)
             }
-            msg @ JoinRequest(_) | msg @ MemberKnowledge { .. } => {
-                debug!(
-                    "{} Unhandled direct message from {}, adding to backlog: {:?}",
-                    self,
-                    p2p_node.public_id(),
-                    msg
-                );
-                self.msg_backlog.push(QueuedMessage::Direct(p2p_node, msg));
+            Variant::BootstrapResponse(_) => {
+                debug!("{} Unhandled message, discarding: {:?}", self, msg);
                 Ok(Transition::Stay)
             }
-            msg => {
-                trace!(
-                    "{} Unhandled direct message {}, ignoring: {:?}",
-                    self,
-                    p2p_node.public_id(),
-                    msg,
-                );
+            _ => {
+                debug!("{} Unhandled message, adding to backlog: {:?}", self, msg);
+                self.msg_backlog.push(msg.into_queued(sender));
                 Ok(Transition::Stay)
             }
         }
     }
 
-    fn handle_hop_message(
+    fn filter_incoming_message(&mut self, message: &MessageWithBytes) -> bool {
+        self.msg_filter.filter_incoming(message).is_new()
+    }
+
+    fn verify_message(&self, msg: &Message) -> Result<bool> {
+        // Only verify messages we know how to handle.
+        match msg.variant {
+            Variant::GenesisUpdate(_)
+            | Variant::Relocate(_)
+            | Variant::MessageSignature(_)
+            | Variant::ParsecRequest(..)
+            | Variant::ParsecResponse(..)
+            | Variant::BootstrapRequest(_)
+            | Variant::ConnectionResponse => (),
+            _ => return Ok(true),
```

Comment:
I'm not really clear why we don't need to verify the other messages.
Also would be good to be exaustive here, what are the message we will not verify.

Reply:
The idea is to verify only the messages we actually handle, because the others might have proof too new for us and we shouldn't fail on that. We are relaying them to our elders anyway and they will do the verification.

Also, this is only really necessary because we have the `log_or_panic` thing in `verify_message`.

Reply:
But makes sense to be exhaustive.

Reply:
If that is the case then it seem that we should do that in should_handle? (though i see that is needed for backlog. It just does not seem reasonable for a verify function to not verify messages. Maybe that verify_message should always return true, and then call it again at the point were we handle it. I'm not sure, but that strange dependency does not seem solid.